In [6]:
import pandas as pd
from pandas import ExcelWriter
from pandas import ExcelFile
#import geopandas as gpd
from shapely.geometry import Point
import shapely
import numpy as np
import matplotlib.pyplot as plt
from scipy import stats
import statsmodels.tsa.stattools as sm
from statsmodels.tsa.arima_model import ARMA
from statsmodels.tsa.arima_model import ARIMA
from statsmodels.tsa.ar_model import AR

/Users/VarshiniSelvadurai/anaconda3/lib/python3.6/importlib/_bootstrap.py:219: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  return f(*args, **kwds)
/Users/VarshiniSelvadurai/anaconda3/lib/python3.6/importlib/_bootstrap.py:219: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  return f(*args, **kwds)
/Users/VarshiniSelvadurai/anaconda3/lib/python3.6/importlib/_bootstrap.py:219: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  return f(*args, **kwds)
/Users/VarshiniSelvadurai/anaconda3/lib/python3.6/importlib/_bootstrap.py:219: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  return f(*args, **kwds)


In [2]:
data = pd.read_excel('/Users/VarshiniSelvadurai/Documents/Internship/Crime_Edited.xlsx', sheet_name = '2017')
#data = pd.read_excel('C:\\Users\\vns8\\Documents\\Crime_Edited.xlsx', sheet_name = '2017')
data = data.loc[data["Crime Name1"] != "Not a Crime"]
data = data.drop(columns = ["Police District Number","Address Number","Victims","CR Number","NIBRS Code","Beat","PRA","State","Incident ID","Offence Code","Sector","Police District Name","Crime Name2","Crime Name3","Agency","Place","End Date/Time","Dispatch Date/Time","Block_Address"])
ugh = list(map(str, data["Start Time"]))
for i in range (0, len(ugh)):
    ugh[i] = ugh[i][:2]
data["Start Time"] = ugh
ugh = (list(map(str,data['Start Date'])))
ughh = []
for i in range(0,len(ugh)):
    ugh[i] = ugh[i][5:]
    ugh[i] = ugh[i][:5]
data['Start Date'] = ugh

In [3]:
data_2018 = pd.read_excel('/Users/VarshiniSelvadurai/Documents/Internship/Crime_Edited.xlsx', sheet_name = '2018')
#data_2018 = pd.read_excel('C:\\Users\\vns8\\Documents\\Crime_Edited.xlsx', sheet_name = '2018')
data_2018 = data_2018.loc[data_2018["Crime Name1"] != "Not a Crime"]
data_2018 = data_2018.drop(columns = ["Police District Number","Address Number","Victims","CR Number","NIBRS Code","Beat","PRA","State","Incident ID","Offence Code","Sector","Police District Name","Crime Name2","Crime Name3","Agency","Place","End Date/Time","Dispatch Date/Time","Block_Address"])
ugh = list(map(str, data_2018["Start Time"]))
for i in range (0, len(ugh)):
    ugh[i] = ugh[i][:2]
data_2018["Start Time"] = ugh
ugh = (list(map(str,data_2018['Start Date'])))
ughh = []
for i in range(0,len(ugh)):
    ugh[i] = ugh[i][5:]
    ugh[i] = ugh[i][:5]
data_2018['Start Date'] = ugh

In [4]:
z_20902 = data.loc[data["Zip Code"] == 20902]
z_20906 = data.loc[data["Zip Code"] == 20906]
z_20904 = data.loc[data["Zip Code"] == 20904]
z_20910 = data.loc[data["Zip Code"] == 20910]
z_20874 = data.loc[data["Zip Code"] == 20874]
z_20902_2018 = data_2018.loc[data_2018["Zip Code"] == 20902]
z_20906_2018 = data_2018.loc[data_2018["Zip Code"] == 20906]
z_20904_2018 = data_2018.loc[data_2018["Zip Code"] == 20904]
z_20910_2018 = data_2018.loc[data_2018["Zip Code"] == 20910]
z_20874_2018 = data_2018.loc[data_2018["Zip Code"] == 20874]

In [5]:
from dateutil import rrule, parser
def daycount(data):
    start_date = list(map(str, data['Start Date'].value_counts().keys().tolist()))
    if (len(start_date)!=365):
        start_date.append("04-23")
    start_date.sort()
    ugh = [0 for i in range(0,len(start_date))]
    for index,row in data.iterrows():
        a = start_date.index(row["Start Date"])
        ugh[a] += 1.0
    start_date = list(rrule.rrule(rrule.DAILY,
                         dtstart=parser.parse("2017-01-01"),
                         until=parser.parse("2017-12-31")))
    for i in range (0,len(start_date)):
        start_date[i] =start_date[i].date()
    march = pd.DataFrame({"Crime Count": ugh, "Date":start_date} )
    #ax = march.plot(kind = "line", y = "Crime Count", figsize = (40,10), title = ("2017" +str(title)), xticks = [0,31,59,90,120,151,181,212,243,273,304,334,365], grid = True)
    #fig = ax.get_figure()
    #fig.savefig("2017" + title + ".png")
    return march

In [6]:
def daycount_2018(data):
    start_date = ['01-01', '01-02', '01-03', '01-04', '01-05', '01-06', '01-07', '01-08', '01-09', '01-10', '01-11', '01-12', '01-13', '01-14', '01-15', '01-16', '01-17', '01-18', '01-19', '01-20', '01-21', '01-22', '01-23', '01-24', '01-25', '01-26', '01-27', '01-28', '01-29', '01-30', '01-31', '02-01', '02-02', '02-03', '02-04', '02-05', '02-06', '02-07', '02-08', '02-09', '02-10', '02-11', '02-12', '02-13', '02-14', '02-15', '02-16', '02-17', '02-18', '02-19', '02-20', '02-21', '02-22', '02-23', '02-24', '02-25', '02-26', '02-27', '02-28', '03-01', '03-02', '03-03', '03-04', '03-05', '03-06', '03-07', '03-08', '03-09', '03-10', '03-11', '03-12', '03-13', '03-14', '03-15', '03-16', '03-17', '03-18', '03-19', '03-20', '03-21', '03-22', '03-23', '03-24', '03-25', '03-26', '03-27', '03-28', '03-29', '03-30', '03-31', '04-01', '04-02', '04-03', '04-04', '04-05', '04-06', '04-07', '04-08', '04-09', '04-10', '04-11', '04-12', '04-13', '04-14', '04-15', '04-16', '04-17', '04-18', '04-19', '04-20', '04-21', '04-22', '04-23', '04-24', '04-25', '04-26', '04-27', '04-28', '04-29', '04-30', '05-01', '05-02', '05-03', '05-04', '05-05', '05-06', '05-07', '05-08', '05-09', '05-10', '05-11', '05-12', '05-13', '05-14', '05-15', '05-16', '05-17', '05-18', '05-19', '05-20', '05-21', '05-22', '05-23', '05-24', '05-25', '05-26', '05-27', '05-28', '05-29', '05-30', '05-31', '06-01', '06-02', '06-03', '06-04', '06-05', '06-06', '06-07', '06-08', '06-09', '06-10', '06-11', '06-12', '06-13', '06-14', '06-15', '06-16', '06-17', '06-18', '06-19', '06-20']
    ugh = [0 for i in range(0,len(start_date))]
    for index,row in data.iterrows():
        a = start_date.index(row["Start Date"])
        ugh[a] += 1.0
    start_date = list(rrule.rrule(rrule.DAILY,
                         dtstart=parser.parse("2018-01-01"),
                         until=parser.parse("2018-06-20")))
    for i in range (0,len(start_date)):
        start_date[i] =start_date[i].date()
    march = pd.DataFrame({"Crime Count": ugh, "Date":start_date} )
    #ax = march.plot(kind = "line", y = "Crime Count", figsize = (40,10), title = ("2017" +str(title)), xticks = [0,31,59,90,120,151,181,212,243,273,304,334,365], grid = True)
    #fig = ax.get_figure()
    #fig.savefig("2017" + title + ".png")
    return march

In [7]:
day_20902 = daycount(z_20902)
day_20906 = daycount(z_20906)
day_20904 = daycount(z_20904)
day_20910 = daycount(z_20910)
day_20874 = daycount(z_20874)
day_20902_2018 = daycount_2018(z_20902_2018)
day_20906_2018 = daycount_2018(z_20906_2018)
day_20904_2018 = daycount_2018(z_20904_2018)
day_20910_2018 = daycount_2018(z_20910_2018)
day_20874_2018 = daycount_2018(z_20874_2018)

In [26]:
zipcode = [day_20906,day_20904,day_20910,day_20874,day_20902]
zipcode_2018 = [day_20906_2018,day_20904_2018,day_20910_2018,day_20874_2018,day_20902_2018]
zipcode_name = [20906,20904,20910,20874,20902]
for i in range (0,5):
    #day_20902_2018.plot()
    print(zipcode_name[i])
    mod = AR(zipcode[i]["Crime Count"])
    print(mod.select_order(ic = "aic",maxlag = 7))
    res = mod.fit(ic = 'aic',maxlag = 7)
    #res.plot_predict(start=365, end=535)
    #plt.show()
    #print(res.summary())
    y = res.predict(start=365, end=535).tolist()
    y_hat = zipcode_2018[i]["Crime Count"].tolist()
    resid = []
    for r in range (0,len(y_hat)):
        resid.append((y_hat[r] - y[r]) * (y_hat[r] - y[r]))
    print("MSE:",np.mean(resid))
    #plt.figure(figsize=(10,5))
    #plt.plot(y_hat)
    #plt.plot(y)
    #plt.show()

20906
7
MSE: 12.401407258520976
20904
1
MSE: 17.540601329487693
20910
1
MSE: 21.307992941106182
20874
7
MSE: 16.908680395074875
20902
1
MSE: 20.816689838049157


20906
7
MSE: 12.401407258520976
20904
1
MSE: 17.540601329487693
20910
1
MSE: 21.307992941106182
20874
7
MSE: 16.908680395074875
20902
1
MSE: 20.816689838049157

In [42]:
zipcode = [day_20906,day_20904,day_20910,day_20874,day_20902]
zipcode_2018 = [day_20906_2018,day_20904_2018,day_20910_2018,day_20874_2018,day_20902_2018]
zipcode_name = [20906,20904,20910,20874,20902]
zipcode_order = [(7,7),(4,4),(4,4),(3,3),(5,4)]
for i in range (0,5):
    #day_20902_2018.plot()
    print(zipcode_name[i])
    mod = ARMA(zipcode[i]["Crime Count"],order = zipcode_order[i])
    print("(p,q):", zipcode_order[i])
    res = mod.fit()
    print("2018")
    y = res.predict(start=365, end=535).tolist()
    y_hat = zipcode_2018[i]["Crime Count"].tolist()
    resid = []
    for r in range (0,len(y_hat)):
        resid.append((y_hat[r] - y[r]) * (y_hat[r] - y[r]))
    print("MSE:",np.mean(resid))
    #plt.figure(figsize=(10,5))
    #plt.plot(y_hat)
    #plt.plot(y)
    #plt.show()
    
    print("2017")
    y_2017 = res.predict(start=0, end=365).tolist()
    y_hat_2017 = zipcode[i]["Crime Count"].tolist()
    resid_2017 = []
    for r in range (0,len(y_hat_2017)):
        resid_2017.append((y_hat_2017[r] - y_2017[r]) * (y_hat_2017[r] - y_2017[r]))
    print("MSE:",np.mean(resid_2017))    
    print()
    #plt.figure(figsize=(10,5))
    #plt.plot(y_hat_2017)
    #plt.plot(y_2017)
    #plt.show()

20906
(p,q): (7, 7)


/Users/VarshiniSelvadurai/anaconda3/lib/python3.6/site-packages/statsmodels/base/model.py:508: ConvergenceWarning: Maximum Likelihood optimization failed to converge. Check mle_retvals
  "Check mle_retvals", ConvergenceWarning)


2018
MSE: 12.269654279677876
2017
MSE: 10.84733131465459

20904
(p,q): (4, 4)


/Users/VarshiniSelvadurai/anaconda3/lib/python3.6/site-packages/statsmodels/base/model.py:488: HessianInversionWarning: Inverting hessian failed, no bse or cov_params available
  'available', HessianInversionWarning)


2018
MSE: 17.494428894296437
2017
MSE: 12.9255576311256

20910
(p,q): (4, 4)
2018
MSE: 22.356365757651478
2017
MSE: 21.81634299338396

20874
(p,q): (3, 3)
2018
MSE: 17.16652529616319
2017
MSE: 13.365355595063075

20902
(p,q): (5, 4)
2018
MSE: 20.56271076138992
2017
MSE: 15.325649124236252



    ugh = []
    ughh = []
    for p in range (0,8):
        for q in range(0,8):
            try:
                mod = ARMA(zipcode[i]["Crime Count"],order = (p,q))
                res = mod.fit()
                #print(p,q)
                #print(res.aic)
                ugh.append((p,q))
                ughh.append(res.aic)
            except:
                continue
20906
(p,q): (7, 7)
AIC: 1937.5631607906917
2018
MSE: 12.269654279677876
2017
MSE: 10.84733131465459

20904
(p,q): (4, 4)
AIC: 1990.6758267523574
2018
MSE: 17.494428894296437
2017
MSE: 12.9255576311256

20910
(p,q): (4, 4)
AIC: 2181.4262537259756
2018
MSE: 22.356365757651478
2017
MSE: 21.81634299338396

20874
(p,q): (3, 3)
AIC: 1998.2222023860309
2018
MSE: 17.16652529616319
2017
MSE: 13.365355595063075
    
20902
(p,q): (5, 4)
AIC: 2054.2187837960632
2018
MSE: 20.56271076138992
2017
MSE: 15.325649124236252

[(7,7),(4,4),(4,4),(3,3),(5,4)]

In [ ]:
zipcode = [day_20906,day_20904,day_20910,day_20874,day_20902]
zipcode_2018 = [day_20906_2018,day_20904_2018,day_20910_2018,day_20874_2018,day_20902_2018]
zipcode_name = [20906,20904,20910,20874,20902]
zipcode_order = [(7,0,7),(4,0,4),(4,0,4),(3,0,3),(5,0,4)]
for i in range (0,5):
    #day_20902_2018.plot()
    print(zipcode_name[i])
    mod = ARIMA(zipcode[i]["Crime Count"],order = zipcode_order[i])
    print("(p,q):", zipcode_order[i])
    res = mod.fit()
    print("2018")
    y = res.predict(start=365, end=535).tolist()
    y_hat = zipcode_2018[i]["Crime Count"].tolist()
    resid = []
    for r in range (0,len(y_hat)):
        resid.append((y_hat[r] - y[r]) * (y_hat[r] - y[r]))
    print("MSE:",np.mean(resid))
    #plt.figure(figsize=(10,5))
    #plt.plot(y_hat)
    #plt.plot(y)
    #plt.show()

zip_order = [(7,0,7),(4,0,4),(4,0,4),(3,0,3),(5,1,6)]
20906
(p,q): (7, 0, 7)
2018
MSE: 12.269654279677876
2017
MSE: 10.84733131465459
    
20904
(p,q): (4, 0, 4)
2018
MSE: 17.494428894296437
2017
MSE: 12.9255576311256

20910
(p,q): (4, 0, 4)
2018
MSE: 22.356365757651478
2017
MSE: 21.81634299338396

20874
(p,q): (3, 0, 3)
2018
MSE: 17.16652529616319
2017
MSE: 13.365355595063075

20902
(p,q): (5, 1, 6)
2018
MSE: 145.0114257311892
2017

In [ ]:
from statsmodels.tsa.statespace.sarimax import SARIMAX
zipcode = [day_20906,day_20904,day_20910,day_20874,day_20902]
zipcode_2018 = [day_20906_2018,day_20904_2018,day_20910_2018,day_20874_2018,day_20902_2018]
zipcode_name = [20906,20904,20910,20874,20902]
for i in range (0,5):
    #day_20902_2018.plot()
    print(zipcode_name[i])
    ugh = []
    ughh = []
    ughhh = []
    for p in range (1,8):
        for d in range (0,8):
            for q in range (0,8):
                for P in range (0,8):
                    for D in range (0,8):
                        for Q in range (0,8):
                            mod = SARIMAX(zipcode[i]["Crime Count"],order=(p,d,q), seasonal_order=(p,d,q,52))
                            res = mod.fit()
                            ugh.append((p,d,q))
                            ughh.append((P,D,Q,52))
                            ughhh.append(res.aic)
    mod = ARIMA(zipcode[i]["Crime Count"],order = ugh[ughhh.index(min(ughhhh))], seasonal_order = ugh[ughhh.index(min(ughhh))])
    print("(p,d,q):",ugh[ughhh.index(min(ughhhh))])
    print("(P,D,Q,52):",ughh[ughhh.index(min(ughhhh))])
    res = mod.fit()
    print("2018")
    y = res.predict(start=365, end=535).tolist()
    y_hat = zipcode_2018[i]["Crime Count"].tolist()
    resid = []
    for r in range (0,len(y_hat)):
        resid.append((y_hat[r] - y[r]) * (y_hat[r] - y[r]))
    print("MSE:",np.mean(resid))
    #plt.figure(figsize=(10,5))
    #plt.plot(y_hat)
    #plt.plot(y)
    #plt.show()

20906


In [ ]:
from statsmodels.tsa.statespace.sarimax import SARIMAX
import warnings
import itertools
y = day_20906["Crime Count"]
p = d = q = range(0, 8)
# Generate all different combinations of p, q and q triplets
pdq = list(itertools.product(p, d, q))
# Generate all different combinations of seasonal p, q and q triplets
seasonal_pdq = [(x[0], x[1], x[2], 52) for x in list(itertools.product(p, d, q))]
warnings.filterwarnings("ignore") # specify to ignore warning messages
ugh = []
ughh = []
ughhh = []
for param in pdq:
    for param_seasonal in seasonal_pdq:
        try:
            mod = SARIMAX(y,
                            order=param,
                            seasonal_order=param_seasonal,
                            enforce_stationarity=False,
                            enforce_invertibility=False)

            results = mod.fit()
            ugh.append(param)
            ughh.append(param_seasonal)
            ughhh.append(results.aic)
            print('ARIMA{}x{}12 - AIC:{}'.format(param, param_seasonal, results.aic))
        except:
            print('ARIMA{}x{}12 - AIC:NOPEEEEEEEE'.format(param, param_seasonal))
print(ugh[ughhh.index(min(ughhh))])
print(ughh[ughhh.index(min(ughhh))])

ARIMA(0, 0, 0)x(0, 0, 0, 52)12 - AIC:2720.1011798296317
ARIMA(0, 0, 0)x(0, 0, 1, 52)12 - AIC:2132.413001087475
ARIMA(0, 0, 0)x(0, 0, 2, 52)12 - AIC:11900.112298434971
ARIMA(0, 0, 0)x(0, 0, 3, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(0, 0, 4, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(0, 0, 5, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(0, 0, 6, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(0, 0, 7, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(0, 1, 0, 52)12 - AIC:1854.7185026003435
ARIMA(0, 0, 0)x(0, 1, 1, 52)12 - AIC:1459.6459764693705
ARIMA(0, 0, 0)x(0, 1, 2, 52)12 - AIC:nan
ARIMA(0, 0, 0)x(0, 1, 3, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(0, 1, 4, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(0, 1, 5, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(0, 1, 6, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(0, 1, 7, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(0, 2, 0, 52)12 - AIC:1805.541680499268
ARIMA(0, 0, 0)x(0, 2, 1, 52)12 - AIC:1313.2314565327074
ARIMA(0, 0, 0)x(0, 2, 2, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(0, 2, 3, 52)12

ARIMA(0, 0, 0)x(2, 5, 0, 52)12 - AIC:nan
ARIMA(0, 0, 0)x(2, 5, 1, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(2, 5, 2, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(2, 5, 3, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(2, 5, 4, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(2, 5, 5, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(2, 5, 6, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(2, 5, 7, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(2, 6, 0, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(2, 6, 1, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(2, 6, 2, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(2, 6, 3, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(2, 6, 4, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(2, 6, 5, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(2, 6, 6, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(2, 6, 7, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(2, 7, 0, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(2, 7, 1, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(2, 7, 2, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(2, 7, 3, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(2, 7, 4, 52)

ARIMA(0, 0, 0)x(5, 2, 0, 52)12 - AIC:-9.976903647532065
ARIMA(0, 0, 0)x(5, 2, 1, 52)12 - AIC:nan
ARIMA(0, 0, 0)x(5, 2, 2, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(5, 2, 3, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(5, 2, 4, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(5, 2, 5, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(5, 2, 6, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(5, 2, 7, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(5, 3, 0, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(5, 3, 1, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(5, 3, 2, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(5, 3, 3, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(5, 3, 4, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(5, 3, 5, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(5, 3, 6, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(5, 3, 7, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(5, 4, 0, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(5, 4, 1, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(5, 4, 2, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(5, 4, 3, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(5, 4,

ARIMA(0, 0, 0)x(7, 7, 0, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(7, 7, 1, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(7, 7, 2, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(7, 7, 3, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(7, 7, 4, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(7, 7, 5, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(7, 7, 6, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 0)x(7, 7, 7, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 1)x(0, 0, 0, 52)12 - AIC:2491.900371291958
ARIMA(0, 0, 1)x(0, 0, 1, 52)12 - AIC:2027.1484837745766
ARIMA(0, 0, 1)x(0, 0, 2, 52)12 - AIC:11127.505977664863
ARIMA(0, 0, 1)x(0, 0, 3, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 1)x(0, 0, 4, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 1)x(0, 0, 5, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 1)x(0, 0, 6, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 1)x(0, 0, 7, 52)12 - AIC:NOPEEEEEEEE
ARIMA(0, 0, 1)x(0, 1, 0, 52)12 - AIC:1848.8362736853137
ARIMA(0, 0, 1)x(0, 1, 1, 52)12 - AIC:1455.5776599449616
ARIMA(0, 0, 1)x(0, 1, 2, 52)12 - AIC:7684.33337484643
ARIMA(0, 0, 1)x(0, 1, 3, 52)12

In [ ]:
from statsmodels.tsa.statespace.sarimax import SARIMAX
import warnings
import itertools
y = day_20904["Crime Count"]
p = d = q = range(0, 8)
# Generate all different combinations of p, q and q triplets
pdq = list(itertools.product(p, d, q))
# Generate all different combinations of seasonal p, q and q triplets
seasonal_pdq = [(x[0], x[1], x[2], 52) for x in list(itertools.product(p, d, q))]
warnings.filterwarnings("ignore") # specify to ignore warning messages
ugh = []
ughh = []
ughhh = []
for param in pdq:
    for param_seasonal in seasonal_pdq:
        try:
            mod = SARIMAX(y,
                            order=param,
                            seasonal_order=param_seasonal,
                            enforce_stationarity=False,
                            enforce_invertibility=False)

            results = mod.fit()
            ugh.append(param)
            ughh.append(param_seasonal)
            ughhh.append(results.aic)
            print('ARIMA{}x{}12 - AIC:{}'.format(param, param_seasonal, results.aic))
        except:
            print('ARIMA{}x{}12 - AIC:NOPEEEEEEEE'.format(param, param_seasonal))
print(ugh[ughhh.index(min(ughhh))])
print(ughh[ughhh.index(min(ughhh))])

In [ ]:
from statsmodels.tsa.statespace.sarimax import SARIMAX
import warnings
import itertools
y = day_20910["Crime Count"]
p = d = q = range(0, 8)
# Generate all different combinations of p, q and q triplets
pdq = list(itertools.product(p, d, q))
# Generate all different combinations of seasonal p, q and q triplets
seasonal_pdq = [(x[0], x[1], x[2], 52) for x in list(itertools.product(p, d, q))]
warnings.filterwarnings("ignore") # specify to ignore warning messages
ugh = []
ughh = []
ughhh = []
for param in pdq:
    for param_seasonal in seasonal_pdq:
        try:
            mod = SARIMAX(y,
                            order=param,
                            seasonal_order=param_seasonal,
                            enforce_stationarity=False,
                            enforce_invertibility=False)

            results = mod.fit()
            ugh.append(param)
            ughh.append(param_seasonal)
            ughhh.append(results.aic)
            print('ARIMA{}x{}12 - AIC:{}'.format(param, param_seasonal, results.aic))
        except:
            print('ARIMA{}x{}12 - AIC:NOPEEEEEEEE'.format(param, param_seasonal))
print(ugh[ughhh.index(min(ughhh))])
print(ughh[ughhh.index(min(ughhh))])

In [1]:
code = input("enter: ")
ugh = code.split()

enter: 12 25 30 32 29 52 51 33 36 26 47 46 297 37 685 90 24 669 684 48 27 49 43 93 111 89 667 670 94 58 110 109 671 95 131 116 108 56 126 59 118 4 18 19 20 23 16 15 31 14 13 358 325 223 230 326 222 313 327 224 328 332 269 367 677 676 678 262 261 341 292 291 276 453 255 253 342 57 99 123 210 218 172 219 163 165 221 355 189 498 431 643 415 390 698 461 388 566 564 642 497 645 495 567 690 494 559 249 528 254 534 343 251 563 452 562 345 592 459 572 580 579 622 633 410 613 611 478 621 477 625 573 503 504 723 585 568 508 493 543 473 623 576 624 610 485 614 480 409 408 404 389 492 689 462 691 680 386 385 521 522 427 510 599 506 630 626 629 609 606 608 616 607 482 499 575 574 490 603 604 615 481 618 628 411 565 718 586 472 600 476 509 601 619 620 28 53 54 44 42 41 40 216 455 460 500 507 496 569 570 706 705 707 704 407 634 635 636 406 405 637 393 398 397 538 537 539 724 639 464 561 588 587 463 584 491 489 581 577 488 717 483 602 349 371 372 531 530 347 644 344 533 526 375 519 679 675 151 146 142

In [3]:
print(len(ugh))

785
